xattr.go

```diff
@@ -43,21 +47,25 @@ type getxattrFunc func(path string, name string, data []byte) (int, error)
 // get contains the buffer allocation logic used by both Get and LGet.
 func get(path string, name string, getxattrFunc getxattrFunc) ([]byte, error) {
 	myname := "xattr.get"
-	// find size.
-	size, err := getxattrFunc(path, name, nil)
-	if err != nil {
-		return nil, &Error{myname, path, name, err}
-	}
-	if size > 0 {
+	// Start with a 1 kiB buffer
+	size := 1024
+	for {
 		data := make([]byte, size)
-		// Read into buffer of that size.
 		read, err := getxattrFunc(path, name, data)
+		if err == syscall.ERANGE || read == size {
```

Comment:
if value is 1K it will read it twice and allocate 2K buffer. I thought about sth.:
```go
for sz := defaultBufSize; sz < maxBufSize; sz <<= 1 {
   size, err := getxattrFunc(path, name, nil)
   if size > sz {
       continue
   }
}
...
```

Reply:
but in my opinion it's the same thing on linux (you just loop faster) and for osx we don't need to check here `read == size`